### ISSUE TITLE
*help* Merge transofrmer with other layers

### ISSUE THREAD

**Issue by Danysolism** [tensorflow/tensor2tensor#762]
Hi, 

I would like to create a new model that uses the transformer encoder and decoder. However, it also needs to include layers in between and the decoder needs to use not only the output of the transformer but also the output of the other layers. Do you have some advice or guide on the best way to do this? Perhaps generate a new T2T model? also, it's not clear for me where would I define the graph of the model? 

Thank you, 
Daniela

**Comment by rsepassi** [comment]
Yes, create your own subclass of `T2TModel`. See the definition of `Transformer` as a guide. You typically only need to override `body`.

**Comment by Danysolism** [comment]
Hi rsepassi,

Thank you for your answer. I created a new T2TModel subclass following the definition of Transformer and decorate it with registry.register_model. I imported it in __init__.py and  I try to run it using the --t2t_usr_dir flag. In my user dir I have the following files: 
`queries.py` (problem definition)
`query_transformer.py` (submodel)
 `__init__.py` (with `from . import queries` and `from . import query_transformer`)
However, I'm getting the following error: `LookupError: Model query_transformer never registered.  Available models.`

Do you happen to know the common errors I could be making or do you have any suggestion of a T2TModel class besides Transformer I could use as guide for my problem?

Thank you.

**Comment by rsepassi** [comment]
please see the example_usr_dir in tensor2tensor/test_data and try to match
its general structure
